### ISSUE TITLE
docker restarting continously after update

### ISSUE THREAD

**Issue by tomse** [inventree/InvenTree#6166]
### Deployment Method

- [ ] Installer
- [ ] Docker Development
- [X] Docker Production
- [ ] Bare metal Development
- [ ] Bare metal Production
- [ ] Digital Ocean image
- [ ] Other (please provide a link `Steps to Reproduce`

### Describe the problem*

running the steps from here https://docs.inventree.org/en/stable/start/docker_prod/#updating-inventree

`docker compose run inventree-server invoke update`

results in error message from git

![image](https://github.com/inventree/InvenTree/assets/1444539/963eb515-159e-4432-ad70-7ab4e58ace98)

### Steps to Reproduce

https://docs.inventree.org/en/stable/start/docker_prod/#updating-inventree

running
`docker compose run inventree-server invoke update`

![image](https://github.com/inventree/InvenTree/assets/1444539/963eb515-159e-4432-ad70-7ab4e58ace98)

if I skip this: and just run
`docker compose up -d`

the worker and server containers keeps restarting
![image](https://github.com/inventree/InvenTree/assets/1444539/fbda9a11-fbfc-44d4-9706-09c366961a23)

**Comment by alimurtaza026** [comment]
I think it is due to moneyed package

**Comment by tomse** [comment]
> I think it is due to moneyed package

?? you get that from the git commands requires 2 parameters error message?

**Comment by matmair** [comment]
We are missing info to really help here. Has this setup ever worked? what image (commit hash + tag) are you using? What are the logs of the restarting containers containing?

This is also the template for installing problems. If you have a problem updating please use a normal bug template as there are a lot of things in inventree instances that can be configured after intial setup that could cause this.

**Comment by alimurtaza026** [comment]
I am using latest.  Master branch

**Comment by alimurtaza026** [comment]
I used docker dev environment to start this,  It's worker and server instance keep starting.  I need it working to fix a bug inn which I assigned

**Comment by tomse** [comment]
> 
> This is also the template for installing problems. If you have a problem updating please use a normal bug template as there are a lot of things in inventree instances that can be configured after intial setup that could cause this.

want me to open a new bug report instead? or should we continue here? in the meantime I'll gather the requested info to the best of my abilities

**Comment by alimurtaza026** [comment]
@tomse , if you can help with it.  That will be great.   You can try setup dev environ using  docker dev

**Comment by matmair** [comment]
@tomse let's just proceed here

**Comment by alimurtaza026** [comment]
Hi guys, can you help me with the setup? Let me know what info you need.  I will provide and proceed with setup?

**Comment by tomse** [comment]
> @tomse let's just proceed here


I had a running setup v12.8 I believe the version was. Running on Ubuntu 22.04.03LTS, installed docker pr the docker website not using Ubuntu's default repo.

I've been using Inventree for a few months, and this is my 3rd upgrade, the 2 first upgrades resultet both in running systems without problems (following the same guide)

I started upgrading to latest docker following this link [Updating inventree](https://docs.inventree.org/en/latest/start/docker_prod/#updating-inventree
)
went into my Inventree/docker/production folder where the docker compose files are stored

ran docker compose down and afterwards pull (basically copy pasting each command from the code tags in the guide.

when I reached the update command
`docker compose run inventree-server invoke update`

I did fix path's etc to match "myuser" rather than default inventree user

```
root@inventree:/home/myuser/InvenTree/docker/production# docker compose run inventree-server invoke update
[+] Creating 2/2
 ✔ Network inventree-production_default           Created                                                          0.1s
 ✔ Container inventree-production-inventree-db-1  Created                                                          0.4s
[+] Running 1/1
 ✔ Container inventree-production-inventree-db-1  Started                                                          0.4s
error: wrong number of arguments, should be 2
usage: git config [<options>]

Config file location
    --global              use global config file
    --system              use system config file
    --local               use repository config file
    --worktree            use per-worktree config file
    -f, --file <file>     use given config file
    --blob <blob-id>      read config from given blob object

Action
    --get                 get value: name [value-pattern]
    --get-all             get all values: key [value-pattern]
    --get-regexp          get values for regexp: name-regex [value-pattern]
    --get-urlmatch        get value specific for the URL: section[.var] URL
    --replace-all         replace all matching variables: name value [value-pattern]
    --add                 add a new variable: name value
    --unset               remove a variable: name [value-pattern]
    --unset-all           remove all matches: name [value-pattern]
    --rename-section      rename section: old-name new-name
    --remove-section      remove a section: name
    -l, --list            list all
    --fixed-value         use string equality when comparing values to 'value-pattern'
    -e, --edit            open an editor
    --get-color           find the color configured: slot [default]
    --get-colorbool       find the color setting: slot [stdout-is-tty]

Type
    -t, --type <type>     value is given this type
    --bool                value is "true" or "false"
    --int                 value is decimal number
    --bool-or-int         value is --bool or --int
    --bool-or-str         value is --bool or string
    --path                value is a path (file or directory name)
    --expiry-date         value is an expiry date

Other
    -z, --null            terminate values with NUL byte
    --name-only           show variable names only
    --includes            respect include directives on lookup
    --show-origin         show origin of config (file, standard input, blob, command line)
    --show-scope          show scope of config (worktree, local, global, system, command)
    --default <value>     with --get, use default value when missing entry

```

this happend. - this error output appears in the default shell.

I'm using portainer to read the logs, but they just show the above error, git requires 2 arguments. - I can supply them if needed, but they do say the same.
to me it seems the init.sh that seems to execute at docker startup has an incomplete git command. but my knowledge of docker images aren't good enough yet to know how to access it and take a look.

let me know what more you need

the docker pull command should fetch the latest stable version

**Comment by tomse** [comment]
> Hi guys, can you help me with the setup? Let me know what info you need. I will provide and proceed with setup?
I think you should start your own bug report or wait to see the outcome of this one, we may have different issues, but posting your problems in this one will only add to the confusion.

**Comment by matmair** [comment]
@tomse are you running on `stable,` `latest` or another tag?

**Comment by tomse** [comment]
@matmair it should be (and looks like) stable

from the .env file:

![image](https://github.com/inventree/InvenTree/assets/1444539/df8eb781-b48d-4062-9579-8fc3ea344e37)

**Comment by SchrodingersGat** [comment]
The only reference to `git config` is here, in the `init.sh` file:

https://github.com/inventree/InvenTree/blob/d1cc81fc9f454f65bbbd40da92e2a27207e52c90/docker/init.sh#L7

If this command is failing for some reason, it would block any further progress.

Do any of the other invoke commands work? Try `docker compose run inventree-server invoke version`

**Comment by matmair** [comment]
I have tried reproducing and have a hard time - did you have any luck @SchrodingersGat?

**Comment by SchrodingersGat** [comment]
No, I have never seen this.

**Comment by tomse** [comment]
@SchrodingersGat I get git error with that one as well

![image](https://github.com/inventree/InvenTree/assets/1444539/260da87e-b987-4c64-bba9-d303ac6fa81f)

just for the sake of it: I've done a git pull in the root of the InvenTree directory and made my custom changes (db user pass etc) to the .env file in `~/InvenTree/docker/production/.env`
I was thinking the init.sh had a bug that was fixed in the latest - hence I did this.

I triede deleting all docker images from the server and do a `docker compose pull`

retried the invoke version command and it throws the same error message "wrong number of arguments"

for the fun of it.. I've tried editing the init.sh in the docker directory just adding some echo lines before and after the first if statement.

I was hoping to see a "1" in the output, but didn't, so init.sh isn't taken from that directory


![image](https://github.com/inventree/InvenTree/assets/1444539/b4763787-93a7-4eb6-9cbc-d08e5ece77db)


![image](https://github.com/inventree/InvenTree/assets/1444539/a4d02be3-fb5e-4cbf-8f11-fd85137fe83a)


I tried the git command running it on the host directly and this shows the same error
so.. really looks like my system doesn't like that command - is the docker image lacking the global config file or something?

![image](https://github.com/inventree/InvenTree/assets/1444539/8d0af511-a8ed-4a7c-b58b-8b4292d48a24)

**Comment by tomse** [comment]
trying to figure out what the git command does I did a quick search, giving me this
https://stackoverflow.com/questions/71849415/i-cannot-add-the-parent-directory-to-safe-directory-in-git

this one shows only 1 wildcard in quotes (single/double) rather than 3 wildcards  
`/home/***`
vs what I would imagine should be..
`"/home/*"`

so I tried running on the host (outside docker):

`git config --global --add safe.directory "/home/*"`

which didn't complain.

**Comment by tomse** [comment]
> The only reference to `git config` is here, in the `init.sh` file:
> 
> https://github.com/inventree/InvenTree/blob/d1cc81fc9f454f65bbbd40da92e2a27207e52c90/docker/init.sh#L7
> 
> If this command is failing for some reason, it would block any further progress.
> 
> Do any of the other invoke commands work? Try `docker compose run inventree-server invoke version`


If I were to test out fixing the int.sh, and build an image to test.. how much of this guide should I follow?
https://docs.inventree.org/en/stable/start/docker_dev/

I'm guessing that's all of the documentation I need to build a production image (bear with me.. still learning docker :-) )

**Comment by SchrodingersGat** [comment]
The docker dev guide is more for developing InvenTree using docker, not so much for the docker image construction itself.

To test locally if the change to `git config` will work, I'd suggest the following steps:

### A. Check that the command fails for the "development" image

1. Pull down your "production" server `docker compose down`
2. `cd` into the root InvenTree directory
3. Build the development image `docker compose build`
4. Check if the same error occurs `docker compose run inventree-dev-server invoke version`

### B. Update init.sh

1. Edit the `docker/init.sh` file accordingly
2. Rebuild the image with `docker compose build

### C. Check again for failure

1. Run `docker compose run inventree-dev-server invoke version`

-------------------

Let me know how it goes!

**Comment by tomse** [comment]
All of this is running directory from the git dir, no modifications to any files (no changes in .env, docker-compose.yaml etc)

Step A.4 after running the invoke version I get this error
![image](https://github.com/inventree/InvenTree/assets/1444539/d8d0b145-57fe-42f9-a042-2b19a3da545b)

if I run 
`docker compose up`
I get this

![image](https://github.com/inventree/InvenTree/assets/1444539/e15d779c-cdd4-4551-ad14-c8fcc932861b)
@alimurtaza026  that sounds like something you mention, if the following doesn't work for you, please make a new bug report rather than try and get support in this thread. make not this is the developer version, remove "-dev" from the command if you use the production server... see it that helps

I ran 
`docker compose run inventree-dev-server invoke update`
which went fine - data being added to the database and nice exit in the end.

`docker compose run inventree-dev-server invoke version`
![image](https://github.com/inventree/InvenTree/assets/1444539/d93e67da-322c-448d-b1b9-120bf3e2f9cb)


`docker compose up`
![image](https://github.com/inventree/InvenTree/assets/1444539/e2c0e745-0615-4cab-98d5-c7bda6cfd0bc)
got a lot of green lines after that.. and at one point it said "press control-c to quit".. so the dev version seems to be working..

soo.. can I do a docker compose build (production)?

**Comment by SchrodingersGat** [comment]
Can you confirm, which of the following is true?

a) You did not modify `init.sh` but it seems to run fine in the dev docker setup - at least it gets past the git errors
b) You did modify `init.sh` and now it works ok in the dev docker setup

**Comment by tomse** [comment]
@SchrodingersGat sorry, yeah the report is a little confusing.

a) **Correct the developer version runs fine** with the unmodified init.sh file. I can see a sign in page on port 8000
b) (to avoid confusion - ignore this answer please) I did modify it adding some debug lines to see where it would stop, but they're non interruptable for the functions of the init.sh.

**Comment by tomse** [comment]
@SchrodingersGat @matmair Thank you very much in trying to assist me. 
I found the error - and it's due to my lack of knowledge of docker and inventree.

the error was that I actually edited the cocker-compose.yaml file.. editing the 2 paths which refered to /data:/home/inventree/data changing them to the paths of "myuser"

I understand that the right side is the path inside the docker.

thanks again, you guys are awesome, and you're created an awesome product (still baffled that it's only version 0.13 - it feels like a released product).
